taichi/ir/type_factory.h

```diff
@@ -23,6 +23,10 @@ class TypeFactory {
 
   Type *get_custom_int_type(int num_bits, bool is_signed);
 
+  Type *get_custom_int_type_with_compute_type(int compute_type_bits,
+                                              int num_bits,
+                                              bool is_signed);
```

Comment:
The function name might be too long. Any good advices would be appreciated!

Reply:
My two cents: actually every `CustomIntType` has a compute type. I would use `get_custom_int_type(int num_bits, bool is_signed, Type *compute_type=nullptr)` if that makes sense to you. Then two APIs can be merged into one.